tests/integration/modules/test_tag.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import pytest
+import saltext.vmware.modules.tag as tagging
+
+
+def test_category_list(patch_salt_globals_tag):
+    """
+    Test create tag functionality
+    """
+    cats = tagging.list_category()
+    if len(cats["categories"]) > 0:
+        assert "urn:vmomi:InventoryServiceCategory" in cats["categories"][0]
+    else:
+        pytest.skip("test requires at least one category")
+
+
+def test_tag_list(patch_salt_globals_tag):
+    """
+    Test list tags functionality
+    """
+    tags = tagging.list_()
+    if len(tags["tags"]) > 0:
+        assert "urn:vmomi:InventoryServiceTag:" in tags["tags"][0]
+    else:
+        pytest.skip("test requires at least one tag")
```

Comment:
**suggestion** similar suggestion here - we should go ahead and create the tags that we expect should be there. In fact, I'd probably just write a single `def test_tagging():` test that looks something like this:

```
def test_tagging():
    tag_names = ['test_foo', 'test_bar', 'test_bang']
    for name in tag_names:
        tag = tagging.get(name)
        tagging.delete(tag)  # or whatever the value is
    # make sure no tags are here
    tags = tagging.list_()
    for existing_tag in tags:
        assert not any(existing_tag["name"] == tag for tag in tag_names)
    # make sure we can create a single tag
    tagging.create(name=tag_names[0], ...)
    tag = tagging.get(tag_names[0])
    assert tag["name"] == tag_names[0]
    # create the rest of the tags
     for tag in tag_names[1:]:
        tagging.create(name=tag, ...)
     tags = tagging.list_()
     # now assert all the expected tags exist
     # now update some tags & test them
     # now delete a tag and assert that only it was deleted and the others are still around
    # now delete the rest of the tags and assert that none of them are here
```

Also as I was writing the above, I realized that we should be using the functional test salt minion. I don't know off the top of my head how to create that, but we do that a ton in salt's tests/pytests/functional/ directory.